### PR TITLE
refactoring to avoid schemas in model

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/met-public/issues/567

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/models/survey.py [hunk 1]
```diff
@@ -32,45 +37,29 @@ class Survey(db.Model):  # pylint: disable=too-few-public-methods
     submissions = db.relationship('Submission', backref='survey', cascade='all, delete')
 
     @classmethod
-    def get_survey(cls, survey_id) -> SurveySchema:
+    def get_survey(cls, survey_id) -> Survey:
         """Get a survey."""
-        survey_schema = SurveySchema()
-        survey = db.session.query(Survey)\
```
Comment (jadmsaadaot) [inline met-api/src/met_api/models/survey.py:38]: do we not need the engagement status data when getting a survey?

Comment (jadmsaadaot) [reply met-api/src/met_api/models/survey.py:38]: We are accessing survye.engagement.status_id in survey building page for example:

    const hasEngagement = Boolean(savedSurvey?.engagement);
    const isEngagementDraft = savedSurvey?.engagement?.status_id === EngagementStatus.Draft;
    const hasPublishedEngagement = hasEngagement && !isEngagementDraft;

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Noticed removal of joins that are necessary

File: met-api/src/met_api/services/survey_service.py [hunk 2]
```diff
@@ -14,48 +16,31 @@ class SurveyService:
...
-    @classmethod
-    def get_unlinked(cls):
-        """Get all surveys."""
-        db_data = Survey.get_all_unlinked_surveys()
-        return db_data
+        survey_model = SurveyModel.get_open(survey_id)
+        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
+        survey = SurveySchema().dump(survey_model)
+        eng = EngagementSchema().dump(engagement_model)
+        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
```
Comment (tiagograf-AOT) [inline met-api/src/met_api/services/survey_service.py:34]: could we maybe move the banner url to a method inside the schema?

**Review by tiagograf-AOT — APPROVED** [review]

File: met-api/src/met_api/models/survey.py (hunk 1 above)
Comment (tiagograf-AOT) [reply met-api/src/met_api/models/survey.py:38]: My understanding is that the engagement is being loaded by a separate call in the service class

Comment (jadmsaadaot) [reply met-api/src/met_api/models/survey.py:38]: const loadSurvey = async () => {
        if (isNaN(Number(surveyId))) {
            navigate('/survey/listing');
            dispatch(
                openNotification({
                    severity: 'error',
                    text: 'The survey id passed was erroneous',
                }),
            );
            return;
        }
        try {
            const loadedSurvey = await getSurvey(Number(surveyId));
            setSavedSurvey(loadedSurvey);
            setName(loadedSurvey.name);
            setLoading(false);
        } catch (error) {
            dispatch(
                openNotification({
                    severity: 'error',
                    text: 'Error occurred while loading saved survey',
                }),
            );
            navigate('/survey/listing');
        }
    };

Comment (jadmsaadaot) [reply met-api/src/met_api/models/survey.py:38]: we're only calling getSurvey

File: met-api/src/met_api/services/survey_service.py (hunk 2 above)
Comment (saravanpa-aot) [reply met-api/src/met_api/services/survey_service.py:35]: I am writing a ticket for improving this.

File: met-api/src/met_api/models/survey.py (hunk 1 above)
Comment (saravanpa-aot) [reply met-api/src/met_api/models/survey.py:39]: I am doing the join at later point of time in the service..
```    @classmethod
    def get(cls, survey_id):
        """Get survey by the id."""
        survey_model: SurveyModel = SurveyModel.get_survey(survey_id)
        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
        survey = SurveySchema().dump(survey_model)
        eng = EngagementSchema().dump(engagement_model)
        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
        survey['engagement'] = eng
        return survey

    @classmethod
    def get_open(cls, survey_id):
        """Get survey by the id."""
        survey_model = SurveyModel.get_open(survey_id)
        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
        survey = SurveySchema().dump(survey_model)
        eng = EngagementSchema().dump(engagement_model)
        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
        survey['engagement'] = eng
        return survey
```

**Review by jadmsaadaot — APPROVED** [review]

**Pull request merged**